zp-core/zp-extensions/downloadList.php

```diff
@@ -424,6 +475,7 @@ static function create($albumname, $fromcache) {
 		}
 		if(!empty($_zp_zip_list)) {
 			$zip = new ZipStream($albumname . '.zip', $opt);
+			zp_setCookie('zpcms_albumzip_ready', 1, 1, null, secureServer());
```

Comment:
This very fast expiring cookie, set at the end of the image list generation for album zip, can be used by JavaScript in themes to detect the starting of a download stream and tell users that the server is managing the download.